ChainSafe/lodestar
# Implement support for validator next-epoch proposer duties

**Motivation**



**Description**

Provides the ability to request block proposers one epoch in the future. See the corresponding issue #3746 for more background information

Closes #3746

**Steps to test or reproduce**

If current Epoch is N,

Make a request to get the proposer duties for epoch N+1:

```
curl -X GET "http://beacon-node/eth/v1/validator/duties/proposer/N+1" -H "accept: application/json"
```

Take note of the returned validator indices.

Using a chain explorer like https://prater.beaconcha.in/, wait till the N+1 epoch becomes the current epoch. Then confirm that the validator indices now listed as part of N+1 are the same returned from the previous curl request to the beacon endpoint.
